Feature/custom numbered list

firmusoft opened the pull request:
Numbering implementation in Custom Style.
Created server for store custom styles globally.

Seybi-ea committed:
fixed couple of  flow errors.

Seybi-ea committed:
1. Fix: Apply the None custom style not removing the applied custom styles
2. Fix: Numbering,indenting,paragraph spacing not reflecting in
on Sample text in Custom style editor.

Seybi-ea committed:
1. FIX: When apply 'None' from style menu, not clearing the applied custom style.
2. Numbering, indenting, paragraph spacing not reflecting on Sample text in Custom Editor UI.

Seybi-ea committed:
Modified the README.md file.

firmusoft committed:
Added new css file to handle paragraph numbering.

firmusoft committed:
Implemented paragraph numbering

firmusoft committed:
Removed text color from number content

firmusoft committed:
Added style to number content.

firmusoft committed:
Fix:Copy/paste custom style numbering.

firmusoft committed:
Handled to apply edited custom style wherever it is used in editor .

Seybi-ea committed:
1. Issue fix : Duplicate style created while modified the style name.

Seybi-ea committed:
Handle enter key (Apply the same style to the new paragraph).

Seybi-ea committed:
Bring custom style for a blank line(commit without completion)

firmusoft committed:
Handled new line style with markson enter key press

Seybi-ea committed:
1. Fix: Underline not working as expected , If user unselect underline
from a custom style it didn't reflect to the editor.
2. Fix: Showing multiple submenus at a time.
3. Fix: Issue fix: sub popup not closing after selecting another(shows
multiple pop ups at a time)

Seybi-ea committed:
1. Fix: Losing focus from the editor when applying a custom Style
2. Fix: Unable to apply a style in a blank line.

Seybi-ea committed:
Handled clear style in empty line

Seybi-ea committed:
Clear marks for empty line on None click

Seybi-ea committed:
Merge conflict

firmusoft committed:
Fix:Capco removes from node on enter key press.

arathygopal-gs committed:
1. Sort stylenames alphabetically
2. Custom style editor UI changes

Seybi-ea committed:
1.Fix:Bullet list not working as expected on indent
2.Fix:Styling issues in style editor

arathygopal-gs committed:
Fix: Selected Text align not showing correctly.

Seybi-ea committed:
1.Added tool tip for style select box

Seybi-ea committed:
Removed apply style from style plugin on document load

Seybi-ea committed:
1.Bold for number part in custom style numbering
2.fixed transaction mismatch error on loading snapshot
3.fixed alphanumeric list issue with bracket.

Seybi-ea committed:
Added functionality for inserting empty elements for custom style numbering when hierachy breaks

firmusoft committed:
Fixed issues in creating empty element for custom style numbering

Seybi-ea committed:
Added blank element below the current selection when heirarchy breaks for numbered style paragraphs.

Seybi-ea committed:
Added indent behaviour for numbered custom style

Seybi-ea committed:
Added option for applying bold for sentence and first word.

arathygopal-gs committed:
1. Server created to handle custom styles globally.
2. Sonar issue fixes.
3. README.md file modifications to include the commands to start custom style server.
4. Clinet side changes related to the new server implementation to save/retireve the custom styles.

arathygopal-gs committed:
Placed a list of existing custom styles in CustomStyle Editor
and populate the styles on clicking on a style in the list

arathygopal-gs committed:
1. FIX: saved custom styles shows in menu only after reloading editor
2. sonar issue fixes
3. FIX: select two paragraph with diffrent style applied,shows the last
applied style insted of None.
4.

Seybi-ea committed:
Issue fix on heirachy when applying none to a numbered custom style

arathygopal-gs committed:
Flow error fixes.

melgish commented on src/StylePlugin.js:
Don't leave commented out code in the project unless you are intending to restore it.  In that case there should be a comment indicating that the code is in transition.

melgish commented on src/ui/CustomMenuUI.js:
Rather verbose way of saying:
```
function  haveEligibleChildren(node, contentLen, styleName) {
  return node.type.name === 'paragraph' && 0 < contentLen && styleName === node.attrs.styleName;
}
```

an if branch can perform slower than an expression under some conditions.

arathygopal-gs committed:
1. Made corrections with respect to the reviewer's comments:
   1. Don't leave commented out code in the project unless you are intending to restore it
   2. Avoid unnecessary if..else statements when returning boolean

firmusoft commented on src/ui/CustomMenuUI.js:
Have made the above mentioned changes in the branch.

arathygopal-gs committed:
Missed to check one condition in Styleplugin.
